include/bout_types.hxx

```diff
@@ -48,21 +48,49 @@ enum REGION {RGN_ALL, RGN_NOBNDRY, RGN_NOX, RGN_NOY, RGN_NOZ};
 
 const std::string& REGION_STRING(REGION region);
 
-/// To identify particular directions (in index space)
-enum class DIRECTION { X = 0, Y = 1, Z = 3, YAligned = 4, YOrthogonal = 5, Special = 6, Null = 7 };
+/// To identify particular directions (in index space):
+///   - X, Y, Z are the coordinate directions
+///   - YAligned is a special case of Y, indicating a field-aligned grid, where
+///     the x- and z- axes are not necessarily orthogonal
+///   - YOrthogonal is a special case of Y, indicating a grid where the x and z
+///     axes are orthogonal but the y-direction is not necessarily
+///     field-aligned
+enum class DIRECTION { X, Y, Z, YAligned, YOrthogonal };
 
 const std::string& DIRECTION_STRING(DIRECTION direction);
 
-void swap(DIRECTION& first, DIRECTION& second);
+/// Identify kind of a field's y-direction
+/// - Standard is the default for the Mesh/Coordinates/ParallelTransform
+/// - Aligned indicates that the field has been transformed to field-aligned
+///   coordinates
+enum class YDirectionType { Standard, Aligned };
```

Comment:
Should this also include something like `Slice` for `FieldPerp`?

Reply:
`FieldPerp` is different depending on whether it's a slice of a standard `Field3D` or a field-aligned one. At the moment it just uses the same `YDirectionTypes` as the `Field3D`. I think that's the simplest way: we could have `Slice` and `SliceAligned` values too and add code to translate/compare those, but I can't think of anywhere we'd need to use or be able to check the difference between `Slice`/`Standard` or `SliceAligned`/`Aligned`. Do you have something in mind?

Reply:
Nothing in particular, just trying to think through everything